packages/theme/src/cli/services/dev.ts

```diff
@@ -53,6 +58,17 @@ export async function dev(options: DevOptions) {
     }, THEME_REFRESH_TIMEOUT_IN_MS)
   }
 
+  if (options['dev-preview']) {
+    outputInfo('This feature is currently in development and is not ready for use or testing yet.')
+    const remoteChecksums = await fetchChecksums(options.theme.id, options.adminSession)
+    const localThemeFileSystem = await mountThemeFileSystem(options.directory)
+    const themeEnvOptions: ThemeEnvironmentOptions = {
+      themeEditorSync: options['theme-editor-sync'],
+    }
+    await startDevServer(options.theme, options.adminSession, remoteChecksums, localThemeFileSystem, themeEnvOptions)
+    return
+  }
```

Comment:
I've created this [proposal PR](https://github.com/Shopify/cli/pull/3852) to clarify and show some details about the Ruby commands. I believe we can skip more Ruby-specific code in the dev preview logic-branch :)

Reply:
thanks for the context + improvements! 
I've incorporated these suggestions + updated tests :)